lib/maths/CBoostedTreeFactory.cc

```diff
@@ -134,6 +134,36 @@ CBoostedTreeFactory::buildForTrain(core::CDataFrame& frame, std::size_t dependen
         new CBoostedTree{frame, m_RecordTrainingState, std::move(treeImpl)}};
 }
 
+CBoostedTreeFactory::TBoostedTreeUPtr
+CBoostedTreeFactory::buildForPredict(core::CDataFrame& frame, std::size_t dependentVariable) {
+
+    m_TreeImpl->m_DependentVariable = dependentVariable;
+
+    skipIfAfter(CBoostedTreeImpl::E_NotInitialized,
+                [&] { this->initializeMissingFeatureMasks(frame); });
+
+    this->prepareDataFrameForTrain(frame);
```

Comment:
This looks surprising. Is this needed for the weight column? I actually wonder if there is a case for having a special setup for predict since the other rows are not needed.